src/SFA.DAS.RoatpOversight.Web.UnitTests/Controllers/Oversight/OversightControllerTests.cs

```diff
@@ -184,5 +197,153 @@ public async Task Post_Outcome_Gateway_Fail_Redirects_To_Confirmed_Page()
+            };
+
+            var result = await _controller.Appeal(request);
+
+            Assert.IsInstanceOf<RedirectToActionResult>(result);
+            var redirectResult = (RedirectToActionResult) result;
+
+            Assert.AreEqual("Outcome", redirectResult.ActionName);
+            Assert.IsTrue(redirectResult.RouteValues.ContainsKey("ApplicationId"));
+        }
+
+        [Test]
+        public async Task Get_Appeal_Preserves_Any_Message_Stored_In_TempData()
+        {
+            var request = new AppealRequest
+            {
+                ApplicationId = Guid.NewGuid()
+            };
+
+            var message = _autoFixture.Create<string>();
+
+            _tempDataDictionary["Message"] = message;
+
+            _appealOrchestrator.Setup(x =>
+                    x.GetAppealViewModel(
+                        It.Is<AppealRequest>(r => r.ApplicationId == request.ApplicationId),
+                        It.Is<string>(m => m == message)))
+                .ReturnsAsync(() => new AppealViewModel());
+
+            var result = await _controller.Appeal(request);
+            Assert.IsInstanceOf<ViewResult>(result);
+        }
+
+        [TestCase(AppealPostRequest.SubmitOption.Upload)]
+        [TestCase(AppealPostRequest.SubmitOption.RemoveFile)]
+        public async Task Post_Appeal_Preserves_Any_Message_In_TempData_On_Post_Of_Non_CTA(AppealPostRequest.SubmitOption submitOption)
+        {
+            var applicationId = Guid.NewGuid();
+            var fileId = Guid.NewGuid();
+
+            _appealOrchestrator.Setup(x =>
+                x.RemoveAppealFile(It.IsAny<Guid>(), It.IsAny<Guid>(), It.IsAny<string>(), It.IsAny<string>()));
+
+            var request = new AppealPostRequest
+            {
+                ApplicationId = applicationId,
+                Message = _autoFixture.Create<string>(),
+                FileUpload = submitOption == AppealPostRequest.SubmitOption.Upload ? GenerateMockFile().Object : null,
+                FileId = submitOption == AppealPostRequest.SubmitOption.RemoveFile ? fileId : Guid.Empty,
+                SelectedOption = submitOption
+            };
+
+            await _controller.Appeal(request);
+
+            Assert.AreEqual(request.Message, _tempDataDictionary["Message"]);
+        }
+
+        private Mock<IFormFile> GenerateMockFile()
+        {
+            var fileMock = new Mock<IFormFile>();
+            var content = _autoFixture.Create<string>();
+            var fileName = "test.pdf";
+            var ms = new MemoryStream();
+            var writer = new StreamWriter(ms);
+            writer.Write(content);
+            writer.Flush();
+            ms.Position = 0;
+            fileMock.Setup(_ => _.OpenReadStream()).Returns(ms);
+            fileMock.Setup(_ => _.FileName).Returns(fileName);
+            fileMock.Setup(_ => _.Length).Returns(ms.Length);
+            return fileMock;
+        }
```

Comment:
There might be a good reason but I can't see why you need to mocking IFormFile when the typical approach would something like this:

`        private static FormFile GenerateFile(string fileName)
        {
            var content = "This is a dummy file";
            return new FormFile(new MemoryStream(Encoding.UTF8.GetBytes(content)), 0, content.Length, fileName, fileName)
            {
                Headers = new HeaderDictionary(),
                ContentType = "application/octet-stream"
            };
        }`

Reply:
No good reason; swapped